### PR TITLE
linuxkit: Fix building on Darwin 

### PR DESCRIPTION
**Pull request by boeyms**
###### Description of changes

Enable linuxkit to build and run on Darwin, including launching of VMs using Apple's Virtualization.framework.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by boeyms** [comment]
Result of `nixpkgs-review pr 216466` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>linuxkit</li>
  </ul>
</details>

**Review by nicknovitski — APPROVED** [review]
I can confirm that this builds and that the resulting binary can `build` an example machine manifest (https://github.com/linuxkit/linuxkit/blob/eb81457111f31c0f5b57e21883a463b7c900399f/linuxkit.yml) and `run` the built image on the macos virtualization framework.

File: pkgs/development/tools/misc/linuxkit/support-apple-11-sdk.patch
Comment (wegank) [inline pkgs/development/tools/misc/linuxkit/support-apple-11-sdk.patch:1]: This patch is huge! Is there any way to avoid putting it in nixpkgs?

Comment (boeyms) [reply pkgs/development/tools/misc/linuxkit/support-apple-11-sdk.patch:1]: @wegank I'll work on raising PR against https://github.com/Code-Hex/vz, but even if it is accepted there, the owners of https://github.com/linuxkit/linuxkit would then have to update the version of github.com/Code-Hex/vz that they vendor in before we could avoid this patch.  (Well, that or the minimum supported SDK for Nixpkgs on Darwin would have to be raised to macOS 12.)

I guess it's up to either yourself or the other maintainers as to whether or not to insist on waiting for if and when that happens.

**Comment by nicknovitski** [comment]
As the maintainer I'm really hoping this can be merged for ZHF (#230712).

File: pkgs/development/tools/misc/linuxkit/default.nix [hunk 1]
```diff
@@ -15,7 +15,14 @@ buildGoModule rec {
 
   modRoot = "./src/cmd/linuxkit";
 
-  buildInputs = lib.optionals stdenv.isDarwin [ Virtualization ];
+  patches = [ ./darwin-os-version.patch ./support-apple-11-sdk.patch ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/linuxkit/default.nix:18]:
```suggestion
  patches = [
    ./darwin-os-version.patch
    ./support-apple-11-sdk.patch
  ];
```

File: pkgs/development/tools/misc/linuxkit/default.nix [hunk 2]
```diff
@@ -25,6 +32,19 @@ buildGoModule rec {
 
   nativeCheckInputs = [ git ];
 
+  # - On macOS, an executable must be signed with the right entitlement(s) to be
+  #   able to use the Virtualization framework at runtime.
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/linuxkit/default.nix:36]: Duplicated comment with above

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 216466` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>linuxkit</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 216466` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>linuxkit</li>
  </ul>
</details>

**Review by RaitoBezarius — APPROVED** [review]

**Review by wegank — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
So late in nixos-22.11 we don't need to backport a patch that fixes a package which was presumably broken for 6 months.

**Pull request merged**